samcli/lib/bootstrap/bootstrap.py

```diff
@@ -83,16 +83,21 @@ def _create_or_get_stack(cloudformation_client):
         raise ManagedStackError(str(ex))
 
 
-def _create_stack(cloudformation_client):
+def _create_stack(cloudformation_client, role_arn=None):
     click.echo("\tCreating the required resources...")
     change_set_name = "InitialCreation"
-    change_set_resp = cloudformation_client.create_change_set(
-        StackName=SAM_CLI_STACK_NAME,
-        TemplateBody=_get_stack_template(),
-        Tags=[{"Key": "ManagedStackSource", "Value": "AwsSamCli"}],
-        ChangeSetType="CREATE",
-        ChangeSetName=change_set_name,  # this must be unique for the stack, but we only create so that's fine
-    )
+
+    params = {
+        "StackName": SAM_CLI_STACK_NAME,
+        "TemplateBody": _get_stack_template(),
+        "Tags": [{"Key": "ManagedStackSource", "Value": "AwsSamCli"}],
+        "ChangeSetType": "CREATE",
+        "ChangeSetName": change_set_name,  # this must be unique for the stack, but we only create so that's fine
+    }
+    if role_arn:
+        params["RoleARN"] = role_arn
```

Comment:
Curious, is setting `RoleARN` to None not work? I would hope that Boto3 or the CloudFormation APIs handle this already, which would allow us to just set it no matter what and remove the need for a if check.

Reply:
Yea it didn’t work with a value of None